### PR TITLE
ice 3.7.0

### PR DESCRIPTION
**Pull request by externl**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by fxcoudert** [comment]
```        == /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/ice.rb ==
C: 24: 24: Space inside square brackets detected.
C: 28:  1: Extra empty line detected at method body beginning.
C: 37: 38: Only use parentheses for ternary expressions with complex conditions.
C: 37: 51: Prefer double-quoted strings inside interpolations.
C: 37: 66: Prefer double-quoted strings inside interpolations.
C: 37: 94: Prefer double-quoted strings inside interpolations.
C: 39: 32: Only use parentheses for ternary expressions with complex conditions.
C: 39: 48: Prefer double-quoted strings inside interpolations.
C: 39: 74: Prefer double-quoted strings inside interpolations.
C: 39:105: Prefer double-quoted strings inside interpolations.
C: 40:  7: Put a comma after the last item of a multiline array.
C: 40: 36: Only use parentheses for ternary expressions with complex conditions.
C: 40: 49: Prefer double-quoted strings inside interpolations.
C: 40: 59: Prefer double-quoted strings inside interpolations.
C: 40: 80: Prefer double-quoted strings inside interpolations.
```

**Comment by externl** [comment]
I'll fix. Why didn't my audit show this?

File: Formula/ice.rb [hunk 1]
```diff
@@ -1,134 +1,89 @@
...
     EOS
     system "#{bin}/slice2cpp", "Hello.ice"
-    system "xcrun", "clang++", "-c", "-I#{include}", "-I.", "Hello.cpp"
-    system "xcrun", "clang++", "-c", "-I#{include}", "-I.", "Test.cpp"
-    system "xcrun", "clang++", "-L#{lib}", "-o", "test", "Test.o", "Hello.o", "-lIce", "-lIceUtil"
-    system "./test", "--Ice.InitPlugins=0"
-    system "/usr/bin/php", "-d", "extension_dir=#{lib}/php/extensions",
-                           "-d", "extension=IcePHP.dy",
-                           "-r", "extension_loaded('ice') ? exit(0) : exit(1);"
+    system "xcrun", "clang++", "-DICE_CPP11_MAPPING", "-std=c++11", "-c", "-I#{include}", "-I.", "Hello.cpp"
```
Comment (ilovezfs) [inline Formula/ice.rb]: Using `xcrun make` in formulae will avoid superenv, which isn't permissible.

Comment (ilovezfs) [inline Formula/ice.rb]: We should not use `xcrun` in the `test do` block.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/ice.rb (hunk 1 above)
Comment (externl) [inline Formula/ice.rb]: What do you propose as an alternative?  `xcrun` handles linking with frameworks and is quite convenient.

Comment (externl) [inline Formula/ice.rb]: @ilovezfs, or is the issue with using compilers in general inside the test block?

Comment (ilovezfs) [inline Formula/ice.rb]: ENV.cc and ENV.cxx

Comment (ilovezfs) [inline Formula/ice.rb]: ENV.cc and ENV.cxx

Comment (externl) [inline Formula/ice.rb]: Can I use `system `xcrun`, ENV.cxx, ...` Otherwise I still have to call `xcrun` to find the sdk location.

Comment (externl) [inline Formula/ice.rb]: I've updated the test. The first part uses just `ENV.cxx`. The second part uses `xcrun` plus `ENV.cxx`. `xcrun` is required to correctly location the macosx sdk location, which is what's being tested.

Comment (externl) [inline Formula/ice.rb]: @ilovezfs Is this okay with you?

Comment (externl) [inline Formula/ice.rb]: @ilovezfs The other option is to remove this part of this test, which I would rather not have to do as it's testing the SDK

Comment (JCount) [inline Formula/ice.rb]: Why is it necessary for testing the SDK that is built? Specifically, why is it not enough to compile against and test `ice`'s SDK in a more standard manner?

**Comment by externl** [comment]
I've fixed the test.

**Comment by fxcoudert** [comment]
Thanks @externl for the pull request!

**Comment by rleigh-codelibre** [comment]
Have you considered making Ice a versioned package?  The current strategy to completely replace the old version is a world breaking change for end users, who can't update their codebases overnight.  This completely breaks packages which were using Ice 3.6.  Now homebrew supports versioning, would it be possible to have `ice` with `ice@3.7` and `ice@3.6` to allow for some measure of stability and continuity of Ice support on MacOS X?

Thanks,
Roger

**Comment by ilovezfs** [comment]
@rleigh-codelibre we could consider it, but given ice had 157 install-on-request events in the last month, it may not meet our guidelines for [acceptable versioned formulae](https://github.com/Homebrew/brew/blob/master/docs/Versions.md#acceptable-versioned-formulae). Are you aware of upstream's `ice36` formula? https://github.com/zeroc-ice/homebrew-tap

**Comment by rleigh-codelibre** [comment]
I wasn't aware, thanks for the pointer!

**Pull request closed without merging**